netdata/netdata
# Update chart's metadata in database when it already exists during creation

##### Summary
When a chart is newly created it is stored in the database along with its metadata. During runtime
any change in the metadata will update the corresponding entry in the database. 

This PR fixes the issue of a chart changing definitions between restarts and having those changes
reflected in the database. 

##### Component Name
database

##### Test Plan
- Start agent with dbengine memory mode (Recommended: disable-cloud)
- Wait for the agent to initialize 
- Stop the agent
- Inspect the database for a specific chart (e.g. cpu.cpufreq)
   ```
   select context from chart where type||"."||id = "cpu.cpufreq" limit 1;
   context        
   ---------------
   cpufreq.cpufreq
   ```
- Stop the agent
   - Modify the collector so that it assigns context `cpu.cpufreq` to that chart
   - Alternatively you can manually update the chart's context e.g. to "bad.context" and observe that it will not be updated
- Restart the agent
   - Notice that the context still remains ` cpufreq.cpufreq` in the database

---
- Apply the PR
- Repeat test to verify that the context has changed in the database
